src/arena.c

```diff
@@ -491,6 +491,20 @@ arena_extent_ralloc_large_expand(tsdn_t *tsdn, arena_t *arena, extent_t *extent,
 	arena_nactive_add(arena, udiff >> LG_PAGE);
 }
 
+static ssize_t
+arena_decay_time_read(arena_t *arena) {
+	arena_decay_time_t dt;
+	dt.u = atomic_read_zu(&arena->decay.time.u);
+	return dt.s;
```

Comment:
Why do you say that using a union like this is undefined behavior?  In the context of type aliasing, unions are pretty much the only well defined mechanism for accessing the same bits as different types.  If anything I'd expect this union approach to be safer than casting (for which unsigned-->signed is technically implementation-defined on overflow).  The down side is that unions prevent a bunch of compiler optimizations which are actually okay, assuming the implementation-defined unsigned-->signed conversion happens to do what we want (i.e. nothing).